bin/check-influxdb-metrics.rb

```diff
@@ -79,32 +79,53 @@ class CheckInfluxDbMetrics < Sensu::Plugin::Check::CLI
          long: '--filter=VALUE',
          description: 'Set the name of your filter, for example: `datacenter`'
 
-  def encode_parameters(parameters)
-    encodedparams = Addressable::URI.escape(parameters)
-    "#{config[:db]}&q=" + encodedparams
-  end
+  option :period,
+         long: '--period=VALUE',
+         description: 'Filter by a given day instead of the last 10 minutes'
 
   def filter_by_environment_when_needed
     config[:tag].nil? && config[:filter].nil? ? '' : " AND \"#{config[:tag]}\" =~ /#{config[:filter]}/"
   end
 
-  def yesterday_query # Reads the value from 10 minutes before yesterday at this time.
-    query = "SELECT sum(\"value\") from \"#{config[:metric]}\" WHERE time > now() - 1455m AND time < now() - 1445m"
+  def base_query
+    "SELECT sum(\"value\") from \"#{config[:metric]}\" "
+  end
+
+  def period
+    config[:period].nil? ? 10 : config[:period].to_i # by default set a period of 10 minutes
+  end
+
+  def today_query_for_a_period
+    start_date = '5'; # starts 5 minutes before now() to let influxdb time to aggregate the data
+    end_date = period + 5; # adds 5 minutes to match with start_date
+    query = query_for_a_period(start_date, end_date)
     query + filter_by_environment_when_needed
   end
 
-  def today_query
-    query = "SELECT sum(\"value\") from \"#{config[:metric]}\" WHERE time > now() - 15m AND time < now() - 5m"
+  def yesterday_query_for_a_period
+    start_date = '1445'; # starts minus 1445 minutes before now() [ yesetrday - 5 minutes] to match with today_query_for_a_period start_date
+    end_date = period + 1445; # adds 1445 minutes to match with start_date
+    query = query_for_a_period(start_date, end_date)
     query + filter_by_environment_when_needed
   end
 
+  def query_for_a_period(start_date, end_date)
```

Comment:
tiny naming niggle: these variables don't contain any `date`s at all, I would just use `start` and `end` in order to avoid confusion